CONTRIBUTING.md

```diff
@@ -11,56 +11,117 @@ Table of Contents:
 
 ### Documentation
 
-Code should be documented where appropriate. The existing code can be used as a guidance here and the general `rustfmt` rules can be followed for formatting.
+Code should be documented where appropriate. The existing code can be used as a guidance here and
+the general `rustfmt` rules can be followed for formatting.
 
-If any change has been made to the `config.rs` file, these changes should also be documented in the example configuration file `alacritty.yml`.
+If any change has been made to the `config.rs` file, these changes should also be documented in the
+example configuration file `alacritty.yml`.
 
-Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to things like code refactorings or documentation/tests) additionally need to be documented in the `CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be used to document changes from a user-perspective, instead of explaining the technical background (like commit messages). More information about Alacritty's change log format can be found [here](https://keepachangelog.com).
+Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to
+things like code refactorings or documentation/tests) additionally need to be documented in the
+`CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be
+used to document changes from a user-perspective, instead of explaining the technical background
+(like commit messages). More information about Alacritty's change log format can be found
+[here](https://keepachangelog.com).
 
 ### Style
 
-All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI build is failing because of formatting issues, you can install rustfmt using `rustup component add rustfmt` and then format all code using `cargo fmt`.
+All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI
+build is failing because of formatting issues, you can install rustfmt using `rustup component add
+rustfmt` and then format all code using `cargo fmt`.
+
+# Release Process
+
+Alacritty's release process aims to provide stable and well tested releases without having to hold
+back new features during the testing period.
+
+To achieve these goals, a new branch is created for every new release. Both the release candidates
+and the final version are only comitted and tagged in this branch. The master branch only tracks
+development versions, allowing us to keep the branches completely separate without merging releases
+back into master.
+
+The exact steps for an exemplary `1.2.3` might look like this:
+ 1. Initially, the version on the latest master is `1.2.3-dev`
+ 2. A new `v1.2.3` branch is created for the release
+ 3. On master, the version is bumped to `1.2.4-dev`
+        and the `-dev` is stripped from previous change log entries
```

Comment:
For maintainers following this in order it would be nice to avoid switching branches many times. This step can be moved below the current step number 7.

Reply:
> This step can be moved below the current step number 7.

No, I specifically don't want to do this. I think it makes sense to separate the two branches as quickly as possible, having a long period of time where the master branch is in an incorrect state does not seem like something we should strive for.

Of course once a contributor is more experienced with the process, they might be able to delay it until the branch is waiting for travis to build for example, which gives a nice timeframe to update master. But generally I think that should be something the dev decides on their own, once they're confident.

> Also, shouldn't we add a new section to the changelog here? We won't know exactly what version number this should be at this time, so maybe this should just be master or something indicating development?

The version is bumped to `1.2.4-dev`, that includes the changelog.